### DISCUSSION TITLE
Health check

### DISCUSSION THREAD

**Discussion by Nintorac** [NVIDIA/NVFlare#4486] [Ideas]
Not sure how this will look like. Total gRPC noob, sorry.

I am trying to stand up a FL server in AWS ECS behind a load balancer, to do so requires the service has a health check and that the health check responds healthy to health probes.

[Here](https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_elasticloadbalancingv2/HealthCheck.html) is the CDK object I need to configure that defines how the health check is performed.

And [here](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) is gRPC docs on health checking.

Is there
a) already a health check, if so how should I configure the cdk Health Check
b) if not already existing is there some workaround that I can use in the meantime

Thanks!

**Comment by Nintorac** [comment]
I have got this working using the `/fedlearn.FederatedTraining/Heartbeat` path and accepting `0-99` response code as a success.

Is there a more specific number/range that indicates a healthy hearbeat or would it be better to implement a specific endpoint for health checks?

**Comment by holgerroth** [comment]
@yhwen, @nvidianz any comments on this one?

**Comment by nvidianz** [comment]
You can simply use TCP as the protocol and just check if the port is open. This works in all cases, even in TLS pass-thru mode. Heartbeat doesn't provide any more information on server's health.

We have plans to add real health check endpoint in the future releases.

**Comment by dean-taylor** [comment]
Is there any update on this?
I was considering adding https://pypi.org/project/gunicorn_worker_healthcheck/

**Reply by holgerroth** [reply]
@chesterxgchen, @pcnudde for viz
